### PR TITLE
Handle twitter misconfiguration error

### PR DESCRIPTION
**Pull request by brianpeiris**
Adds error handling to twitter API calls, so that we react accordingly instead of crashing the game loop.

Goes with https://github.com/mozilla/reticulum/pull/543

### CODE REVIEW

File: src/react-components/room/TwitterOAuthModalContainer.js [hunk 1]
```diff
@@ -30,7 +30,14 @@ export function TwitterOAuthModalContainer({ hubChannel, onConnected, onClose })
         popup.focus();
         popupRef.current = popup;
       } catch (error) {
-        console.error(error);
+        if (error === "twitter_probably_misconfigured") {
```
Comment (brianpeiris) [inline src/react-components/room/TwitterOAuthModalContainer.js:33]: This seems wrong now that I look at it. `error` should probably be an `Error` object, not a string.

File: src/react-components/room/TwitterOAuthModalContainer.js [hunk 2]
```diff
@@ -30,7 +30,14 @@ export function TwitterOAuthModalContainer({ hubChannel, onConnected, onClose })
...
+          console.error(
+            "Twitter might be misconfigured. Check the docs here: https://hubs.mozilla.com/docs/hubs-cloud-enable-media-browser.html#twitter"
```
Comment (brianpeiris) [inline src/react-components/room/TwitterOAuthModalContainer.js:35]: We might want to fix this URL in the docs before we merge this?

File: src/utils/hub-channel.js [hunk 3]
```diff
@@ -360,7 +360,7 @@ export default class HubChannel extends EventTarget {
         .receive("ok", res => {
           resolve(res.oauth_url);
         })
-        .receive("error", reject);
+        .receive("error", err => reject(err.reason));
```
Comment (robinkwilson) [inline src/utils/hub-channel.js:363]: Should we pass in the error object?

File: src/react-components/room/TwitterOAuthModalContainer.js (hunk 1 above)
Comment (robinkwilson) [reply src/react-components/room/TwitterOAuthModalContainer.js]: I can change the way we handle the error if this still looks wrong.

**Pull request merged**